### DISCUSSION TITLE
Support Steam launcher?

### DISCUSSION THREAD

**Discussion by furai** [Heroic-Games-Launcher/HeroicGamesLauncher#1330] [Ideas]
Hey,

Just an idea for people on different hardware than steam deck - would it be possible to integrate steam as well?

Cheers,
Furai

**Comment by PaffcioStudio** [comment]
I approve, I just have concerns about whether Steam will want to make competition. And even if it does, the problem could be access to workshops and DLC. There will be a lot of work to do. No less, I am ready to give up all launchers and use only Heroic.

**Comment by l3org** [comment]
I too would like to see Steam support in Heroic Launcher.

I am so, so sick and tired of having to put up with Valve's messy Steam client that is sluggish, has zero accessibility, no theme support and bloated software that eats way too much memory. The list goes on, but I'm sure we can all relate on this matter. If you guys can make it happen, please, please make it so. I am so sick of dealing with Steam.

I would even donate to such a cause if there were a donations page or something set up. That is how bad I want to break away from the clutches of Valve's platform software.

**Comment by Alex0007** [comment]
In addition to all replies above – Steam doesn't let you download Windows games on Mac (to run them via CrossOver). Yes, i'm trying to avoid it, but sometimes devs don't care about alternative stores

**Comment by Subtly-Dismiss** [comment]
This would be an excellent addition

**Comment by wiiznokes** [comment]
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/issues/2872

**Comment by l3org** [comment]
So I came across this discussion again after a long while of not visiting it and figured I'd give everyone an update to break the bad news to you all.

I spoke with them on their discord a long while ago (back in 2023) and they said they absolutely will not support Steam - period. Shortly after they banned me for offering donation support to try make it happen.

Great community. /s

If you want a decent software to run your games, use Bottles. It supports the mainstream launchers unlike Heroic Launcher. The community is a lot nicer, less toxic and supportive as well.

**Reply by Lariaa** [reply]
It comes down to necessity and complexity.
Steam is available for Windows, Linux and MacOS directly so regardless of your operating system you can have a native build.
This is not true for Epic, AmazonGames, GoG or most other launchers/storefronts.
This ranks Steam very low in terms of necessity.

In addition to that you have the problem of complexity. Steam has a lot of features that need to be carefully reverse engineered and rebuild in open source similiarly to the CLI tools we base Heroic on (legendary, gogdl, nile etc.)
This is a huge amount of work required and you need someone that can do it and will do it.
At the moment there are other, lower hanging fruits that can easily be done first which we are working on.

This by no means is a final death sentence to your idea. If someone is willing to work on reversing Steam and build a proper backend for it and wants to integrate this with Heroic, we will always welcome contributions.
It is just not something the current contributors will work on in the near future.

**Reply by l3org** [reply]
Ideas are welcome? Ha! You were offered contributions which would have lead to being able to pay someone who may have been interested in contributing their efforts. You declined. At least be honest. You guys aren't going to work on it at all.

**Reply by termdisc** [reply]
You were not banned because of your request or donation offer for Steam support.

You were banned because you started insulting the developers for having "dumb ideas" while you were criticizing experimental UI stuff. You were asked to be more polite, and you went off the rails to be insulting and disrespectful to anyone that was in the server.

**Reply by CommandMC** [reply]
Hello,
this being a year old & you being banned, I unfortunately no longer have the full context on your interactions in the server. From what I can gather from the context: 

Your initial outlook was that implementing Steam should be prioritized & implementing the core featureset (which I've told you back then already, is different for everyone) isn't hard to do. We then explained that priorities aren't really a thing, everyone works on what they want to work on (and given Steam runs on everything, it's not a likely target for people to work on); if you really want something to be done *right now*, you'll have to work on that yourself (or find someone to work on it for you).  

You then came back a couple days later, personally attacking the author of some perceived UI change you've noticed. I do not want to put words into your mouth, but community members responded with "you come off hyper aggressive and entitled" and "it certainly isn't polite and respectful". One quoted the start of your message, reading "just want to say that whoever's idea it was ...". After repeatedly being told to calm down, you did not comply and were banned because of it.

At no point do I see mention of contributions coming from your side, financial or otherwise, but again I'm only reading messages responding/reacting to yours, so this may have been missed. It would seem rather unlikely to me though that someone verbally attacking us also offers to give us money.

Please do correct me if anything here is not accurate. As it stands, I do not see why this ban is unjustified. If you think that your behavior has changed since this incident, I'd be open to giving you another chance.

**Reply by l3org** [reply]
> You were banned because you started insulting the developers for having "dumb ideas" while you were criticizing experimental UI stuff.

Of course you would paint everyone else to be the good guy and smear me as the bad one. Gotta keep that +PR. appearance.

**Reply by l3org** [reply]
> Hello, this being a year old & you being banned, I unfortunately no longer have the full context on your interactions in the server. From what I can gather from the context:
> 
> Your initial outlook was that implementing Steam should be prioritized & implementing the core featureset (which I've told you back then already, is different for everyone) isn't hard to do. We then explained that priorities aren't really a thing, everyone works on what they want to work on (and given Steam runs on everything, it's not a likely target for people to work on); if you really want something to be done _right now_, you'll have to work on that yourself (or find someone to work on it for you).
> 
> You then came back a couple days later, personally attacking the author of some perceived UI change you've noticed. I do not want to put words into your mouth, but community members responded with "you come off hyper aggressive and entitled" and "it certainly isn't polite and respectful". One quoted the start of your message, reading "just want to say that whoever's idea it was ...". After repeatedly being told to calm down, you did not comply and were banned because of it.
> 
> At no point do I see mention of contributions coming from your side, financial or otherwise, but again I'm only reading messages responding/reacting to yours, so this may have been missed. It would seem rather unlikely to me though that someone verbally attacking us also offers to give us money.
> 
> Please do correct me if anything here is not accurate. As it stands, I do not see why this ban is unjustified. If you think that your behavior has changed since this incident, I'd be open to giving you another chance.

Of course you don't have that 'full context' anymore. Convenient. You too have to keep that +PR appearance.

There's a reason you guys have so few online members anymore and it's not because of people like me. People out here can see for themselves just by that alone, how things operate around there. I never said anything should be prioritized at all. Fabrication upon fabrication. Where does it end? I also never implied that anything had to be rushed. That's yet another fabrication.

As for 'attacking' the dev, call it what you want - an attack, a jab, return fire (which is what I prefer to call it), etc., if someone's going to blatantly disrespect me, they are going to get both barrels shot right back at them, but conveniently, I'm painted as the bad guy so you guys can paint the pretty picture you guys want the general public to see. You saw no contributions from my side because I was a new member around there and you were told this repeatedly. I was trying to be involved, but you guys weren't having it because you didn't like to discuss Steam as it was according to all the admin and freedom fighters on Discord, 'an old topic we're sick and tired of talking about'.

Your members (specifically admins//mods/devs) are excessively defensive and arrogant. It's not my fault they can't discuss without being offended at every line of text that comes their way that they don't like to see or agree with. Everything must be praise around there or GTFO. It's as simple as that. This is why your guys' discord is a ghost town compared to what it once was. The way the HL team operates things and people are treated around there - it's no wonder there's only a handful of people who are usually online, contrasted to the massive amount of offline members that never come/came back.

Your last line in closing says a lot too. That's basically your way of jabbing a knife into my liver from behind, before closing your statements. If my attitude has changed you'd be open to giving me another chance? Well, now I and everyone else out here knows who the knife bearers are. It's not me or my attitude that needs changing. I won't ever be back to your discord and anyone who values their time and energy whom also has self respect, won't subject themselves to what goes on around there or be involved with Heroic Launcher at all.

**Reply by CommandMC** [reply]
> Of course you don't have that 'full context' anymore. Convenient. You too have to keep that +PR appearance.

As you were banned, your messages were deleted. This is standard practice for messages containing hateful content.

> I never said anything should be prioritized at all. Fabrication upon fabrication

As I've explained, this was what I was able to gather from the context around your messages & my personal recollection. I did not state everything there was exactly how it happened, and as I've mentioned, was open for corrections. I, quite fortunately, talk to a lot of people, so I unfortunately cannot recall every detail of every conversation, especially if it happened a year ago.

> I also never implied that anything had to be rushed. That's yet another fabrication.

Where did I accuse you of that? My explanation of priorities (I assume the "if you really want something to be done right now" tripped you up) was speaking in a general tone. As I've mentioned repeatedly now, I do not know what exactly your messages contained.

> if someone's going to blatantly disrespect me, they are going to get both barrels shot right back at them

And where exactly did that happen? You came in yelling at people about a change, where were you disrespected?
I have records from everything other than your messages, and I know when your messages happened. There was no conversation including/about you before that.

> I'm painted as the bad guy so you guys can paint the pretty picture you guys want the general public to see

The quotes I've included in my reply are... quotes. The messages are there. Anyone not banned can go and read them. That is how the situation happened.

> I was trying to be involved, but you guys weren't having it because you didn't like to discuss Steam as it was according to all the admin and freedom fighters on Discord, 'an old topic we're sick and tired of talking about'.

The 1000th person asking "Steam support when?" is definitely an old topic. People discussing how to implement Steam's features, maybe even sharing proof of concepts, is very much appreciated. From the records I have, it seems that your messages fell under the former rather than the latter category however.

It should be noted that it was first explained to you why implementing Steam's full featureset might detract contributors from working on it. Only after you explained that this task was "not too hard" did you receive a few less-neutral messages. I will quote them here, as I do not think they qualify to be as negative as you described:
- "you are welcome to reimplement all that 😉"
- "I'm pretty sure that an oss alternative would already exist if it was not too hard"

You've also gotten one comment that *might* fall under that description, although it was made by a regular user:
- "you think that if someone were to reverse engineer Steam, all of it, and reimplement all of its functionality, that this would somehow be superior to actual Steam?"

> Your members (specifically admins//mods/devs) are excessively defensive and arrogant. It's not my fault they can't discuss without being offended at every line of text that comes their way that they don't like to see or agree with

Sorry, but you can't excuse personally attacking people with "people are too sensitive". You were told to not do that & to calm down, you didn't, you got banned. Do the same thing in any other server (including Bottles BTW!), you'll get the same outcome.

> Everything must be praise around there or GTFO

Legitimate criticism is very much appreciated. "This person is an idiot for making that change" (paraphrased) is not. You may still *think* that to yourself, but do not spew nonsense and negativity like that on our primary communication platform.

> This is why your guys' discord is a ghost town compared to what it once was. The way the HL team operates things and people are treated around there - it's no wonder there's only a handful of people who are usually online, contrasted to the massive amount of offline members that never come/came back.

Our Discord is growing at a very normal pace. We have recently surpassed 15k members, around 5k of which joined just this year, with 2.5k currently being online. The amount of messages in a given timeframe has also increased, albeit slightly. Further than that, Discord sadly does not give many tools to monitor engagement, otherwise I'd be happy to present them here.
The suggestion that our Discord is not active is, frankly, laughable.

I am also wondering how you're monitoring server activity, given you are banned.

> If my attitude has changed you'd be open to giving me another chance?

I did not think it had to be clarified, but yes, if you stop attacking people & being aggressive for no reason, you would be welcome to participate again.

> I won't ever be back to your discord and anyone who values their time and energy whom also has self respect, won't subject themselves to what goes on around there or be involved with Heroic Launcher at all.

Unfortunate. Given you've repeatedly now both misrepresented the original situation and misinterpreted my comments about it, I also retract my previous offer of giving you another chance.

I wish you good luck with Bottles, please send Mirko my regards. I look forward to meeting him at this year's Ubuntu Summit & watching his Vanilla OS talk, and hope he drops by our Linux Gaming booth. Although having GE and the creators of Lutris, Heroic, and Bottles all in one place might be too much for the universe to handle.

**Comment by imLinguin** [comment]
Closing this thread now, if someone wishes to add anything constructive to the topic of Steam support in Heroic please do so in #2872
